src/electrum_wallet_file.rs

```diff
@@ -0,0 +1,438 @@
+                        )
+                    }
+
+                    fn visit_str<E>(self, value: &str) -> Result<Field, E>
+                    where
+                        E: de::Error,
+                    {
+                        let re = Regex::new(r#"(x)(\d+)(/)|([a-z_\-0-9]+)"#).unwrap();
+                        let captures = re.captures(value).map(|captures| {
+                            captures
+                                .iter()
+                                .skip(1)
+                                .flatten()
+                                .map(|c| c.as_str())
+                                .collect::<Vec<_>>()
+                        });
+                        match captures.as_deref() {
+                            Some(["x", _i, "/"]) => Ok(Field::Keyst),
+                            Some(["keystore"]) => Ok(Field::Keyst),
+                            Some(["addresses"]) => Ok(Field::Addrs),
+                            Some(["wallet_type"]) => Ok(Field::WalTyp),
+                            _ => Ok(Field::Ignore),
+                        }
+                    }
+                }
+
+                deserializer.deserialize_identifier(FieldVisitor)
+            }
+        }
+
+        struct ElectrumWalletFileVisitor;
+
+        impl<'de> de::Visitor<'de> for ElectrumWalletFileVisitor {
+            type Value = ElectrumWalletFile;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("struct ElectrumWalletFile")
+            }
+
+            fn visit_map<V>(self, mut map: V) -> Result<ElectrumWalletFile, V::Error>
+            where
+                V: de::MapAccess<'de>,
+            {
+                let mut addresses = Addresses::new();
+                let mut keystores = Vec::new();
+                let mut wallet_type = WalletType::Standard;
+
+                while let Some(key) = map.next_key()? {
+                    match key {
+                        Field::Addrs => {
+                            addresses = map.next_value()?;
+                        }
+                        Field::Keyst => {
+                            keystores.push(map.next_value()?);
+                        }
+                        Field::WalTyp => {
+                            wallet_type = map.next_value()?;
+                        }
+                        Field::Ignore => {
+                            let _ignore = map.next_value::<de::IgnoredAny>()?;
+                        }
+                    }
+                }
+
+                Ok(ElectrumWalletFile {
+                    addresses,
+                    keystores,
+                    wallet_type,
+                })
+            }
+        }
+
+        const FIELDS: &[&str] = &[
+            "addresses",
+            "addr_history",
+            "channel_backups",
+            "keystore",
+            "wallet_type",
+            "x1/",
+            "x2/",
+            "x3/",
```

Comment:
does this mean x4, x5... are not parsed?

Reply:
As I understand it, this is just a hint for use with error messages. I picked just a couple of typical fields.

Reply:
Ok, thanks for the explanation